src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,533 @@
+  // {top_left.x + rc_width - rc_height * 2 / 20,                              // pt2 = ..., ...
+  //  top_left.y + rc_height * 1 / 20},
+  // {top_left.x + rc_width, top_left.y + rc_height * 10 / 20},                // pt3 = ..., ...
+  // {top_left.x + rc_width - rc_height * 2 / 20,                              // pt4 = pt2.x, ...
+  //  top_left.y + rc_height * 19 / 20},                                       
+  // {top_left.x + rc_width - rc_height * 4 / 20, top_left.y + rc_height},     // pt5 = pt1.X, ... 
+  // {top_left.x + rc_height * 5 / 20, top_left.y + rc_height},                // pt6 = pt0.x, pt5.y
+  // {top_left.x + rc_height * 2 / 20, top_left.y + rc_height * 19 / 20},      // pt7 = ..., pt4.y
+  // {top_left.x, top_left.y + rc_height * 10 / 20},                           // pt8 = ..., pt3.y
+  // {top_left.x + rc_height * 2 / 20, top_left.y + rc_height * 1 / 20}};      // pt9 = pt7.x, pt2.y                                                // pt9 = pt7.x, pt2.y
+
+  canvas.Select(look.frame_pen);
+  canvas.Select(look.background_brush);
+  canvas.DrawPolygon( polygone_frame, ARRAY_SIZE(polygone_frame));
+}
+
+
+void 
+NavigatorRenderer::DrawText(Canvas &canvas, const Waypoint &wp_current, 
+                            const PixelRect &rc,[[maybe_unused]] const NavigatorLook &look, bool inverse) noexcept{
+  const auto &basic = CommonInterface::Basic();
+  const auto &calculated = CommonInterface::Calculated();
+  
+  bool has_started = calculated.ordered_task_stats.start.task_started;
+
+  // PREPARE TEXTS
+  const auto time_elapsed = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_elapsed}};
+  const RoughTimeDelta r{};
+
+  auto time_elapsed_s = FormatLocalTimeHHMM(time_elapsed, r).c_str();
+  has_started ? time_elapsed_s : time_elapsed_s = "--:--";
+
+  const auto time_start = calculated.ordered_task_stats.start.time;
+  auto time_start_s = FormatLocalTimeHHMM(time_start,
+                                          CommonInterface::GetComputerSettings().utc_offset).c_str();
+  has_started ? time_start_s : time_start_s = "--:--";
+
+  const auto time_local_s = FormatLocalTimeHHMM(basic.time,
+                                                CommonInterface::GetComputerSettings().utc_offset).c_str();
+
+  const auto time_planned = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_planned}};
+  auto time_planned_s = FormatLocalTimeHHMM(time_planned, r).c_str();
+  has_started ? time_planned_s : time_planned_s = "--:--";
+
+  const auto arrival_planned = TimeStamp{FloatDuration{time_start.ToDuration() + time_planned.ToDuration()}};
+  auto arrival_planned_s = FormatLocalTimeHHMM(arrival_planned, 
+                                               CommonInterface::GetComputerSettings().utc_offset).c_str(); 
+  has_started ? arrival_planned_s : arrival_planned_s = "--:--";
+  
+  // e_WP_Name
+  TCHAR waypoint_name_s[40];
+  strncpy(waypoint_name_s, wp_current.name.c_str(), ARRAY_SIZE(waypoint_name_s)-1);
```

Comment:
Codacy is right - `strncpy()` is a hazard and it looks like you don't know what it does. Don't use it unless you know what you're doing, aware of all the pitfalls.